core/domain/exp_jobs_one_off.py

```diff
@@ -128,19 +142,29 @@ def entity_classes_to_map_over(cls):
 
     @staticmethod
     def map(item):
+        is_valid_math_expression = schema_utils.get_validator(
+            'is_valid_math_expression')
+        is_valid_math_equation = schema_utils.get_validator(
+            'is_valid_math_equation')
+        ltt = latex2text.LatexNodes2Text()
```

Comment:
> I thought we were planning to validate the asciimath, not the latex.

Currently the input is stored in the form of latex which is why we need to convert it before validating it. Also, when we perform the actual upgrade, we would have to replace the latex string with the text format to maintain consistency.

> Also, is this library meant to be a permanent addition?

No. Once the upgradation of the old explorations has been done successfully, we can remove this from the codebase.

> What is the precise definition of the format of the text output of latex2text? Is it asciimath? If so, could you point me to the docs which state that? If not, then wouldn't this mean that there is a mismatch?

The [documentation](https://pylatexenc.readthedocs.io/en/latest/latex2text/) doesn't mention the exact format of the text output with the exact grammar; but I've played around with this locally and it seems to convert the latex syntax into human readable string. So things like `\frac{x}{y}` turns into `x/y` and so on. Note that there is no tight requirement for having asciimath as long as it is in a human-readable form. The reason for choosing asciimath was that it was the closest to a human-readable form that guppy outputted in,

Reply:
I'm afraid I don't follow. A MathExpression has an ascii property and a latex property (look at the definition of the object in extensions/objects). If I understand correctly, the former is supposed to be in asciimath. So why don't you just validate that and avoid latex conversion altogether?

Reply:
Ah, I understand what the confusion is. The thing is, the expected answer in this interaction is indeed in the form of an object that contains both, latex as well as asciimath. But, this is only for the learner's answer. The creator's input is in Unicode because while creating an exploration with this interaction, the creator gets a normal text input box instead of the guppy editor. And obviously, the text input cannot output both latex as well as asciimath. Here, we are just expecting the creator to enter a correctly formatted latex string without any validations to check if the string entered is correctly formatted. I'm not sure what the thought process was behind not using the guppy editor in the creator's view as well, but because of this, we don't have the asciimath format of the creator's input, which is what this job validates. Does that clear things up?

Reply:
Yes, this makes sense, thanks for the clear explanation. Just to check -- the new interactions will be using guppy input, and the stored format will be math expression input with both ascii and latex fields, right? (And we'll be migrating from the old unicodestring to the new mathexpression format?) If so, all that sounds good.

(I'm not sure why the unicode editor was used originally, but I agree that it's definitely suboptimal.)

Reply:
Yes, the new interactions will all use guppy for the learner and the creator's views. The answer will be stored as `AlgebraicExpression`, `MathEquation`, or `NumericExpression`. But only asciimath format from the guppy editor will be stored since we don't need latex. And yes, all the old latex input strings will be converted to ascii and will be stored in the new format.

Reply:
OK, that sounds good, thanks.

Just to check, would it be possible to convert from asciimath to latex, if we ever wanted to do so? (If so, how?)

Reply:
Yes, pylatexenc also has a reversed functionality to convert text to latex. Also, if we ever need latex in the frontend, nerdamer has functionality to convert the expression/equation to latex.

Reply:
OK, sounds good. Thanks!